### ISSUE TITLE
Looping lightray inside a dataset

### ISSUE THREAD

**Issue by Anikbh11** [trident-project/trident#116]
Hi,

Is it possible to loop a light ray n times over inside a single dataset until a redshift interval of dz is achieved? I mean is there a way I can input the specific dataset (like simple light ray) and loop my sightline n times in it? 

I'm using a very small boxsize and need a light ray that traverses a much larger (10x-15x) redshift interval. 

Thanks, 
Aniket

**Comment by chummels** [comment]
`make_compound_ray()` was designed for just this reason.

https://trident.readthedocs.io/en/latest/annotated_example.html#compound-lightrays

**Comment by Anikbh11** [comment]
Hi Cameron, 

Thanks for your response. 

I tried using the GIZMO snapshots that are linked in the trident example section to test if I can loop inside a single box. The gizmo folder that I downloaded contains 3 snapshots (snap_N128L16_149.hdf5, 150 and 151 which are at z = 0.1, 0.005 and 0 respectively.) To test if I can loop inside the same box to achieve very long sightlines I removed the 150 and 151 snapshots keeping only the 149 one in the directory. The size of the box from what I can see is 16 cMpc which corresponds to a dz of ~0.004 at z = 0.1. So for example, if I want to calculate a sightline with dz = 0.1 (from z = 0.1 to 0), I need to loop across the box 25 times. Here's what I do in trident : 

ray = trident.make_compound_ray('N128L16.param',simulation_type='Gadget',near_redshift=0.0,far_redshift=0.01,max_box_fraction=25)

with the max_box_fraction = 25, I figured I would get a ray that loops the necessary 25 times inside the .149 box. But trident raises an error : 

YTOutputNotIdentified: Supplied ('snap_N128L16_151.hdf5',) {}, but could not load!

What am I doing wrong? Can I achieve looping within a single box like this? 

For context, I want to apply a similar thing to a set of snapshots that have a frequency of dz = 0.3. Therefore I won't have snapshots nearby enough and I need to loop over inside a single box.
